mcs/class/System/Mono.AppleTls/AppleTlsContext.cs

```diff
@@ -487,23 +498,23 @@ public void SetClientSideAuthenticate (SslAuthenticate auth)
 				IntPtr id;
 				var result = SSLGetPeerID (Handle, out id, out length);
 				CheckStatusAndThrow (result);
-				if ((result != SslStatus.Success) || (length == 0))
+				if ((result != SslStatus.Success) || ((int)length == 0))
 					return null;
-				var data = new byte [length];
+				var data = new byte [(int)length];
 				Marshal.Copy (id, data, 0, (int) length);
 				return data;
 			}
 			set {
 				SslStatus result;
-				nint length = (value == null) ? 0 : value.Length;
+				nint length = (value == null) ? (IntPtr)0 : (IntPtr)value.Length;
```

Comment:
Good point. When I originally did this I expected to need nint everywhere, but it was just this one file.

I'm converting them all to IntPtr.